File: app/src/main/java/org/aossie/agoraandroid/adapters/VoterRecyclerAdapter.java
package org.aossie.agoraandroid.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import org.aossie.agoraandroid.R;

public class VoterRecyclerAdapter
    extends RecyclerView.Adapter<VoterRecyclerAdapter.VoterViewHolder> {
  private final ArrayList<String> mVoterNameList;
  private final ArrayList<String> mVoterEmailList;

  public VoterRecyclerAdapter(ArrayList<String> mVoterNameList, ArrayList<String> mVoterEmailList) {
    this.mVoterNameList = mVoterNameList;
    this.mVoterEmailList = mVoterEmailList;
  }

  @NonNull
  @Override
  public VoterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    LayoutInflater li = (LayoutInflater) parent.getContext()
        .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    View voterView = li.inflate(R.layout.list_item_voter_details, parent, false);
    return new VoterRecyclerAdapter.VoterViewHolder(voterView);
  }

  @Override
  public void onBindViewHolder(@NonNull VoterViewHolder holder, int position) {
    holder.voterName.setText(mVoterNameList.get(position));
    holder.voterEmail.setText(mVoterEmailList.get(position));
  }

  @Override
  public int getItemCount() {
    return mVoterEmailList.size();
  }

  static class VoterViewHolder extends RecyclerView.ViewHolder {
    final TextView voterName, voterEmail;
    final View itemView;

    public VoterViewHolder(@NonNull View itemView) {
      super(itemView);
      voterName = itemView.findViewById(R.id.text_view_voter_name);
      voterEmail = itemView.findViewById(R.id.text_view_voters_email);
      this.itemView = itemView;
    }
  }
}

File: app/src/main/java/org/aossie/agoraandroid/adapters/VoterRecyclerAdapter.kt
package org.aossie.agoraandroid.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import org.aossie.agoraandroid.R.id
import org.aossie.agoraandroid.R.layout
import org.aossie.agoraandroid.adapters.VoterRecyclerAdapter.VoterViewHolder
import org.aossie.agoraandroid.data.remote.dto.VotersDto

class VoterRecyclerAdapter(
  private val votersList: List<VotersDto>,
) : Adapter<VoterViewHolder>() {

  override fun onCreateViewHolder(
    parent: ViewGroup,
    viewType: Int
  ): VoterViewHolder {
    val li = parent.context
      .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    val view = li.inflate(layout.list_item_voter_details, parent, false)
    return VoterViewHolder(view)
  }

  override fun onBindViewHolder(
    holder: VoterViewHolder,
    position: Int
  ) {
    holder.voterName.text = votersList.get(position).voterName
    holder.voterEmail.text = votersList.get(position).voterEmail
  }

  override fun getItemCount(): Int {
    return votersList.size
  }

  class VoterViewHolder(itemView: View) : ViewHolder(itemView) {
    val voterName: TextView = itemView.findViewById(id.text_view_voter_name)
    val voterEmail: TextView = itemView.findViewById(id.text_view_voters_email)
  }
}
